src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -0,0 +1,698 @@
+        emit RewardClaimed(address(FXS), totalFxs_, block.timestamp);
+    }
+
+    function swapCrvForWeth(uint256 _crvAmount) internal {
+        // Swap CRV -> WETH
+        CRV_ETH_CURVE_POOL.exchange(
+            1, 0, _crvAmount, (getCrvAmountInEth(_crvAmount) * minOutBpsCrvToWeth.val) / MAX_BPS
+        );
+    }
+
+    function swapCvxForWeth(uint256 _cvxAmount) internal {
+        // Swap CVX -> WETH
+        CVX_ETH_CURVE_POOL.exchange(
+            1, 0, _cvxAmount, (getCvxAmountInEth(_cvxAmount) * minOutBpsCrvToWeth.val) / MAX_BPS
+        );
+    }
+
+    function swapWethForDai() internal returns (uint256 daiEarned_) {
+        // Swap WETH -> DAI
+        uint256 wethBalance = WETH.balanceOf(address(this));
+        daiEarned_ = UNIV3_ROUTER.exactInputSingle(
+            IUniswapRouterV3.ExactInputSingleParams({
+                tokenIn: address(WETH),
+                tokenOut: address(DAI),
+                fee: uint24(500),
+                recipient: owner(),
+                deadline: type(uint256).max,
+                amountIn: wethBalance,
+                amountOutMinimum: (getWethAmountInDai(wethBalance) * minOutBpsWethToUsdc.val) / MAX_BPS,
+                sqrtPriceLimitX96: 0 // Inactive param
+            })
+        );
+    }
+
+    function swapFxsForDai(uint256 _fxsAmount) internal returns (uint256 daiEarned_) {
+        address[] memory path = new address[](2);
+        path[0] = address(FXS);
+        path[1] = address(FRAX);
+        // 1. Swap FXS -> FRAX
+        uint256[] memory amounts = FRAXSWAP_ROUTER.swapExactTokensForTokens(
+            _fxsAmount,
+            (getFxsAmountInFrax(_fxsAmount) * minOutBpsFxsToFrax.val) / MAX_BPS,
+            path,
+            address(this),
+            block.timestamp
+        );
+        uint256 fraxBal = amounts[amounts.length - 1];
+        // 2. Swap FRAX -> DAI
+        daiEarned_ = FRAX_3CRV_CURVE_POOL.exchange_underlying(
+            0, 1, fraxBal, (getFraxAmountInDai(fraxBal) * minOutBpsFraxToDai.val) / MAX_BPS, owner()
+        );
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC VIEW
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Checks whether an upkeep is to be performed.
+    /// @dev The calldata is encoded with the `processRewardsKeeper` selector. This selector is ignored when
+    ///      performing upkeeps using the `performUpkeep` function.
+    /// @return upkeepNeeded_ A boolean indicating whether an upkeep is to be performed.
+    /// @return performData_ The calldata to be passed to the upkeep function.
+    function checkUpkeep(bytes calldata) external override returns (bool upkeepNeeded_, bytes memory performData_) {
+        uint256 crvPending;
+        uint256 fxsPending;
+        uint256 length = baseRewardPools.length();
+
+        for (uint256 i = 0; i < length;) {
```

Comment:
```suggestion
        for (uint256 i; i < length;) {
```

Reply:
added the suggestion across the contract 8f1285e8799cd5cb413b8cb597f1829926afab18